packages/primitives/src/Switch/Switch.tsx

```diff
@@ -0,0 +1,122 @@
+import * as React from 'react'
+import { SxStyleProp } from 'theme-ui'
+import { Box, Input, Label } from '../'
+
+interface SwitchProps extends React.InputHTMLAttributes<HTMLInputElement> {
+  label?: string
+  inline?: boolean
+  displayInForm?: boolean
+  sx?: SxStyleProp
+  variant?: string
+  variantKey?: string
+}
+
+type SwitchIndicatorProps = {
+  checked?: boolean
+  disabled?: boolean
+} & Pick<SwitchProps, 'sx' | 'variant' | 'variantKey'>
+
+function SwitchIndicator({
+  checked,
+  disabled,
+  sx,
+  variant,
+  variantKey,
+}: SwitchIndicatorProps) {
+  return (
+    <Box
+      variant={`${variant}.${checked ? 'checked' : 'unchecked'}`}
+      variantKey={variantKey}
+      as="span"
+      borderStyle="solid"
+      borderWidth={1}
+      opacity={disabled ? 0.7 : 1}
+      borderRadius={40}
+      display="inline-block"
+      height={16}
+      ml={-38}
+      mr={2}
+      position="relative"
+      sx={{
+        transition: 'background-color 0.1s cubic-bezier(0.4, 1, 0.75, 0.9)',
+        ...(disabled && { cursor: 'not-allowed', outline: 'none' }),
+
+        'input:focus ~ &': {
+          outlineWidth: 2,
+          outlineStyle: 'solid',
+          outlineOffset: 2,
+        },
+
+        '&::before': {
+          borderRadius: '50%',
+          content: '""',
+          display: 'block',
+          height: 12,
+          width: 12,
+          position: 'absolute',
+          top: '50%',
+          left: checked ? 'calc(100% - 13px)' : '1px',
+          transform: 'translateY(-50%)',
+          transition: 'left 0.1s cubic-bezier(0.4, 1, 0.75, 0.9)',
+        },
+        ...sx,
+      }}
+      width={28}
+    />
+  )
+}
+
+export const Switch = React.forwardRef(
+  (
+    {
+      label,
+      inline = false,
```

Comment:
@alexanderchan let's chat about this when you get back next week, I think `inline` should probably move to a variant. If we do this, we should be the `switch` variant to the root of the theme object instead of under forms so that you can have a default and `switch.inline` variant. I think that I would like to lean towards always using variants for how a component is rendered within the page layout. The same could go for `displayInForm` if it is a valid enough use case. This also helps reduce developer confusion by knowing that they need to just add a variant anytime they want to modify how the component is rendered versus going and looking up which props will handle what they want to do.

Reply:
Please push back if you don't think this is a good idea.